### PR TITLE
docs: added subcommands docs page

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
## Issue

N/A

## Description

This PR adds a documentation page that goes into greater detail about each subcommand.

### CODE REVIEW

File: docs/subcommands.md [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+
+> [!WARNING]
+>
+> Credentials and other permissions may be required depending on the Validator plugins you use. For example, the AWS
+> plugin requires AWS credentials with elevated permissions to validate your AWS environment.
+
+## Install
+
+Use the `install` subcommand to install the Validator framework and configure Validator plugins. An interactive wizard
+will guide you through the installation process. You can also use a configuration file to install the Validator.
```
Comment (TylerGillson) [inline docs/subcommands.md:21]: nit: please use just "Validator" everywhere vs. "the Validator"

File: docs/subcommands.md [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+| -------------- | --------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------- | -------- |
+| `-f`           | `--config-file` | Install the Validator using a configuration file (optional). Provide the file path to the configuration file.                                              | string   |
+| `-o`           | `--config-only` | Generate a configuration file without proceeding with an actual install. Default: false                                                                    | boolean  |
+| `-h`           | `--help`        | Help with any command.                                                                                                                                     | -        |
+| `-r`           | `--reconfigure` | Reconfigure Validator and plugins prior to installation. The `--config-file` flag must be included. Default: false.                                        | boolean  |
```
Comment (TylerGillson) [inline docs/subcommands.md:36]: Could you please add an example below for `--reconfigure`? 🙏🏼

File: docs/subcommands.md [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+| `-p`           | `--password`    | Update credentials provided in the configuration file. This does not proceed with installation. The `--config-file` flag must be included. Default: false. | boolean  |
```
Comment (TylerGillson) [inline docs/subcommands.md:37]: Should be `--update-passwords`

File: docs/subcommands.md [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+The `validator` command exposes the following subcommands.
+
+- [`describe`](#describe) - Describe the Validator results in a Kubernetes cluster.
+
+- [`install`](#install) - Install the Validator framework and configure Validator plugins.
+
+- [`uninstall`](#uninstall) - Uninstall the Validator framework and remove all Validator plugins.
+
+- [`upgrade`](#upgrade) - Upgrade Validator & reconfigure validator plugins.
+
```
Comment (TylerGillson) [inline docs/subcommands.md:12]: Can we please document the `version` subcommand too?

File: docs/subcommands.md [hunk 5]
```diff
@@ -0,0 +1,437 @@
...
+```shell
+validator install --config-only
+```
+
+Update credentials provided in the configuration file. This does proceed with installation but will prompt for new
+credentials.
+
+```shell
+validator install --password --config-file /Users/demo/.validator/validator-20231109135306/validator.yaml
```
Comment (TylerGillson) [inline docs/subcommands.md:66]: `--update-passwords`

File: docs/subcommands.md [hunk 6]
```diff
@@ -0,0 +1,437 @@
...
+
+## Install
+
+Use the `install` subcommand to install the Validator framework and configure Validator plugins. An interactive wizard
+will guide you through the installation process. You can also use a configuration file to install the Validator.
+
+> [!NOTE]
+>
+> A [kind](https://kind.sigs.k8s.io/) cluster will be deployed as part of the Validator installation. The name of the kind
+> cluster is `validator-kind-cluster`. You can install the Validator into an existing Kubernetes cluster by using the Helm chart. Refer to the
```
Comment (TylerGillson) [inline docs/subcommands.md:26]: `validatorctl` will also install validator into an existing cluster if you have the KUBECONFIG env var set. This is recommended.

**Review by TylerGillson — COMMENTED** [review]
Thank you @karl-cardenas-coding! This is awesome and much appreciated. I regret to mention that some significant changes to these commands will be happening soon in https://github.com/validator-labs/validator/issues/294.

**Comment by karl-cardenas-coding** [comment]
@TylerGillson Thanks for the feedback. I went ahead and pushed up the changes :)

**Review by TylerGillson — APPROVED** [review]

**Comment by TylerGillson** [comment]
fyi @karl-cardenas-coding tests are breaking due to the version bump enforced via `make reviewable`. @ahmad-ibra is working on the fix in a separate PR. We'll update and merge this one as soon as his changes to properly upgrade the validator version are in.

**Comment by ahmad-ibra** [comment]
@karl-cardenas-coding heads up that we'll need to wait for https://github.com/validator-labs/validatorctl/pull/115 to be merged before this PR goes in. Once the PR i linked is merged the test failures we see in CI will be addressed.

**Comment by karl-cardenas-coding** [comment]
All good gents 👍

**Pull request merged**